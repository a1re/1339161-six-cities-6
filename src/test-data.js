export const testOfferList = [
  {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 1
  },
  {
    "city": {
      "name": `Cologne`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 3
  },
  {
    "city": {
      "name": `Brussels`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 4
  },
  {
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 5
  },
  {
    "city": {
      "name": `Moscow`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 6
  },
  {
    "city": {
      "name": `Hamburg`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 9
  },
  {
    "city": {
      "name": `Dusseldorf`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "preview_image": ``,
    "images": [],
    "title": ``,
    "is_favorite": true,
    "is_premium": true,
    "rating": 0,
    "type": `hotel`,
    "bedrooms": 0,
    "max_adults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "is_pro": true,
      "avatar_url": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 7
  }
];

export const adaptedCityList = [
  {
    name: `Paris`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  }, {
    name: `Cologne`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  }, {
    name: `Brussels`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  }, {
    name: `Amsterdam`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  }, {
    name: `Hamburg`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  }, {
    name: `Dusseldorf`,
    location: {
      latitude: 0,
      longitude: 0,
      zoom: 0
    }
  },
];

export const adaptedOfferList = [
  {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 1
  },
  {
    "city": {
      "name": `Cologne`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 3
  },
  {
    "city": {
      "name": `Brussels`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 4
  },
  {
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 5
  },
  {
    "city": {
      "name": `Moscow`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 6
  },
  {
    "city": {
      "name": `Hamburg`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 9
  },
  {
    "city": {
      "name": `Dusseldorf`,
      "location": {
        "latitude": 0,
        "longitude": 0,
        "zoom": 0
      }
    },
    "previewImage": ``,
    "images": [],
    "title": ``,
    "isFavorite": true,
    "isPremium": true,
    "rating": 0,
    "type": `Hotel`,
    "bedrooms": 0,
    "maxAdults": 0,
    "price": 0,
    "goods": [],
    "host": {
      "id": 25,
      "name": ``,
      "isPro": true,
      "avatarUrl": ``
    },
    "description": ``,
    "location": {
      "latitude": 0,
      "longitude": 0,
      "zoom": 0
    },
    "id": 7
  }
];

export const testReviewList = [
  {
    "id": 1,
    "user": {
      "id": 1,
      "is_pro": true,
      "name": ``,
      "avatar_url": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  },
  {
    "id": 2,
    "user": {
      "id": 1,
      "is_pro": true,
      "name": ``,
      "avatar_url": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  },
  {
    "id": 3,
    "user": {
      "id": 1,
      "is_pro": true,
      "name": ``,
      "avatar_url": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  }
];

export const adaptedReviewList = [
  {
    "id": 1,
    "user": {
      "id": 1,
      "isPro": true,
      "name": ``,
      "avatarUrl": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  },
  {
    "id": 2,
    "user": {
      "id": 1,
      "isPro": true,
      "name": ``,
      "avatarUrl": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  },
  {
    "id": 3,
    "user": {
      "id": 1,
      "isPro": true,
      "name": ``,
      "avatarUrl": ``
    },
    "rating": 0,
    "comment": ``,
    "date": `1970-01-01T00:00:00Z`
  }
];

export const testUserInfo = {
  "id": 1,
  "email": `test@domain.org`,
  "name": ``,
  "avatar_url": ``,
  "is_pro": true
};

export const adaptedUserInfo = {
  "id": 1,
  "email": `test@domain.org`,
  "name": ``,
  "avatarUrl": ``,
  "isPro": true
};

export const testReview = {
  "id": 4,
  "user": {
    "id": 1,
    "is_pro": true,
    "name": ``,
    "avatar_url": ``
  },
  "rating": 0,
  "comment": ``,
  "date": `1970-01-01T00:00:00Z`
};

export const adaptedReview = {
  "id": 4,
  "user": {
    "id": 1,
    "isPro": true,
    "name": ``,
    "avatarUrl": ``
  },
  "rating": 0,
  "comment": ``,
  "date": `1970-01-01T00:00:00Z`
};
