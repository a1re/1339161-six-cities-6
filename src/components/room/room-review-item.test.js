import React from 'react';
import {render} from '@testing-library/react';
import RoomReviewItem from './room-review-item';
import {adaptedReviewList} from '../../test-data';

it(`Room review item renders correctly`, () => {
  const {container} = render(<RoomReviewItem review={adaptedReviewList[0]}/>);
  expect(container).toMatchSnapshot();
});
