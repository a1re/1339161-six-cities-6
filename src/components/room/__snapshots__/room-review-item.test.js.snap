// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room review item renders correctly 1`] = `
<div>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Kendall"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Kendall
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width: 80%;"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
      </p>
      <time
        class="reviews__time"
        datetime="2021-02-21T08:04:28.647Z"
      >
        February 2021
      </time>
    </div>
  </li>
</div>
`;
